### PR TITLE
Migrate demos with external plugins to new installation methods

### PR DESCRIPTION
**Pull request by f1ames**
This PR migrates demos with 3rd-party plugins to NIM. Follow-up of #38:

> I can prepare separate branch migrating all demos with external plugins which will be waiting to be merged. So we have still all demos fully functional (and fully featured) on `master` and then will be ready when those external plugins are migrated.

It's based on `new-installation-methods-v2` branch (#38 PR) and I followed all the same conventions and assumptions. You can read #38 description for all details.

The main difference here is that 3rd-party plugins (WProofreader and Mathtype) are commented out in CKEditor 5 plugins list in it's config in each demo. This way demos can be run and after plugins are migrated to NIM, it should be enough to just uncomment those lines.

### CODE REVIEW

File: feature-rich/index.ts [hunk 1]
```diff
@@ -487,513 +511,507 @@ const EMOJIS_ARRAY = [
...
+				'|',
+				'importWord',
+				'exportWord',
+				'exportPdf',
+				'|',
+				'findAndReplace',
+				'selectAll',
+				'wproofreader',
+				'|',
+				// 'insertTemplate',
```
Comment (Reinmar) [inline feature-rich/index.ts:603]: I've just tripped on this. I provided the license key, as it asked me to, but then surprise, surprise: those premium buttons weren't there. I had quite a big WTH moment :D 

The good thing about our toolbar initialization is that even if you used a button that wasn't registered, the editor will work and you will get a warning.  
  
So it's not necessary to comment out missing buttons. And it's definitely confusing :P

**Comment by Reinmar** [comment]
Why did you rewrite the demos to TS?

My worry here: So far, we keep our resources in JS (docs and most other samples I can think of).

**Comment by Reinmar** [comment]
8f66b1f is a bit ugly commit but I tried to figure out why the demos don't work

Turned out there were two reasons: dup modules because of WSC and the change in CSS file names that we did recently.

**Comment by Reinmar** [comment]
I started pushing here changes that we have to do to make the cke5-demos repo follow changes that were implemented on ckeditor.com.

**Comment by f1ames** [comment]
> Why did you rewrite the demos to TS?
> 
> My worry here: So far, we keep our resources in JS (docs and most other samples I can think of).

It was a part of this task as we initially discussed with @Witoso and @arkflpc. But I don't know if there is a longer story behind this, maybe they can give more context.

> [8f66b1f](https://github.com/ckeditor/ckeditor5-demos/commit/8f66b1fa7e473007527bf77c8df084426f2cf71e) is a bit ugly commit but I tried to figure out why the demos don't work
> 
> Turned out there were two reasons: dup modules because of WSC and the change in CSS file names that we did recently.

Yeah, I pushed it before the CSS file names changes, also @pszczesniak pointed that out in the first PR - https://github.com/ckeditor/ckeditor5-demos/pull/38#pullrequestreview-2101619442. I see you already updated it 🙇 👍 

> I started pushing here changes that we have to do to make the cke5-demos repo follow changes that were implemented on ckeditor.com.

@Reinmar maybe let me know what you are working on here, and how cna I help so we don't duplicate the work.

**Comment by f1ames** [comment]
Ready for review.

The only thing missing is enabling MathType and WProofReader in couple of demos, but this needs to wait for NIM compatible versions of those plugins.

File: headless/package.json [hunk 2]
```diff
@@ -1,59 +1,25 @@
...
-		"build-dev": "webpack --mode development",
-		"start": "webpack serve --open /index.html --mode development"
+		"dev": "vite --open",
+		"build": "vite build",
+		"preview": "vite preview"
+	},
+	"devDependencies": {
+		"@ckeditor/ckeditor5-react": "alpha",
+		"ckeditor5": "alpha",
+		"ckeditor5-premium-features": "alpha",
```
Comment (pszczesniak) [inline headless/package.json:20]: We can remove `ckeditor5-premium-features` dependency cause it's not used in this demo.

Comment (pszczesniak) [reply headless/package.json]: Addressed in 8a36a76.

**Review by pszczesniak — APPROVED** [review]
LGTM :+1:

File: headless/package.json (hunk 2 above)
Comment (f1ames) [reply headless/package.json]: Good catch, thanks for the patch 👍

**Comment by pszczesniak** [comment]
@f1ames i've tested `WProofreader` with alpha version `"@webspellchecker/wproofreader-ckeditor5": "3.0.0-alpha.0"` and it works ok.

**Pull request merged**